cmd/image-builder/build.go

```diff
@@ -4,31 +4,34 @@ import (
 	"fmt"
 	"os"
 	"path/filepath"
+	"strings"
 
 	"github.com/osbuild/bootc-image-builder/bib/pkg/progress"
 	"github.com/osbuild/images/pkg/imagefilter"
 )
 
 type buildOptions struct {
-	OutputDir string
-	StoreDir  string
+	OutputDir      string
+	StoreDir       string
+	OutputBasename string
 
 	WriteManifest bool
 	WriteBuildlog bool
 }
 
-func buildImage(pbar progress.ProgressBar, res *imagefilter.Result, osbuildManifest []byte, opts *buildOptions) error {
+func buildImage(pbar progress.ProgressBar, res *imagefilter.Result, osbuildManifest []byte, opts *buildOptions) (string, error) {
 	if opts == nil {
 		opts = &buildOptions{}
 	}
 
+	basename := basenameFor(res, opts.OutputBasename)
```

Comment:
This seems to slightly mishandle basenames which already contain dots in them, for example:

```
user@muja image-builder-cli € ll output 
total 488396
-rw-r--r--. 1 root root    315185 Mar 14 12:21 Fedora-Minimal-42-20250314.buildroot-build.spdx.json
-rw-r--r--. 1 root root    750529 Mar 14 12:21 Fedora-Minimal-42-20250314.image-os.spdx.json
-rw-r--r--. 1 root root     79595 Mar 14 12:23 Fedora-Minimal-42-20250314.n.1.buildlog
-rw-r--r--. 1 root root    153945 Mar 14 12:21 Fedora-Minimal-42-20250314.n.1.osbuild-manifest.json
-rw-r--r--. 1 root root 498810732 Mar 14 12:23 Fedora-Minimal-42-20250314.n.1.raw.xz
```